R/archive.R

```diff
@@ -145,14 +150,64 @@ epi_archive =
                     c("geo_type", "time_type"))) {
               Warn("`additional_metadata` names overlap with existing metadata fields \"geo_type\", \"time_type\".")
             }
-
+            
+            # Finish off with compactify
+            if (missing(compactify)) {
+              compactify = NULL
+            } else if (!rlang::is_bool(compactify) &&
+                        !rlang::is_null(compactify)) {
+              Abort("compactify must be boolean or null.")
+            }
+            
             # Create the data table; if x was an un-keyed data.table itself,
             # then the call to as.data.table() will fail to set keys, so we
             # need to check this, then do it manually if needed
             key_vars = c("geo_value", "time_value", other_keys, "version")
             DT = as.data.table(x, key = key_vars)
             if (!identical(key_vars, key(DT))) setkeyv(DT, cols = key_vars)
-
+            
+            # Checks to see if a value in a vector is LOCF
+            is_locf <- function(vec) {
+              ifelse(!is.na(vec) & !is.na(lag(vec)),
+                     vec == lag(vec),
+                     is.na(vec) & is.na(lag(vec)))
+            }
+            
+            # Checks for LOCF's in a data frame
+            rm_locf <- function(df) {
+             filter(df,if_any(c(everything(),-version),~ !is_locf(.))) 
+            }
+            
+            # Keeps LOCF values, such as to be printed
+            keep_locf <- function(df) {
+              filter(df,if_all(c(everything(),-version),~ is_locf(.))) 
+            }
+            
+            # Runs compactify on data frame
+            if (is.null(compactify) || compactify == TRUE) {
+              elim = keep_locf(DT)
+              DT = rm_locf(DT)
+            } else {
+              # Create empty data frame for nrow(elim) to be 0
+              elim = tibble()
+            }
+            
+            # Warns about redundant rows
+            if (is.null(compactify) && nrow(elim) > 0) {
+              warn <- paste("\nLOCF rows found. To remove warning, set",
+              "compactify to TRUE or fix these rows: \n")
```

Comment:
New users are likely to see this the first time they try to make an epi archive; this message may not be explanatory enough for them.  Maybe re-use some of the `@param` documentation.  "Fix" might be misleading; I think the most common thing is just that many recording mechanisms would naturally have some or a lot of redundant rows, and if those were never generated or read it might make the entire pipeline more efficient; or, maybe they can't be removed upstream and they can be removed here for space efficiency; or, probably less likely, these redundant rows are faulty records and there's some bug upstream the data pipeline.

Reply:
I replaced the word "fix." Perhaps a good idea would be to put on the vignette about what may be inherently wrong with having LOCF rows.

Reply:
I'm tweaking this message and some of the documentation.  Then I will try to make that version of the documentation true by addressing #109.  Then I think this will be good to go!